ui/index.d.ts

```diff
@@ -152,6 +151,18 @@ export interface AzExtServiceClientCredentials {
      * @returns {Promise<WebResourceLike>} The signed request object;
      */
     signRequest(webResource: any): Promise<any>;
+    
+    /**
+     * Gets the token provided by this credential.
+     *
+     * This method is called automatically by Azure SDK client libraries. You may call this method
+     * directly, but you must also handle token caching and token refreshing.
+     *
+     * @param scopes - The list of scopes for which the token will have access.
+     * @param options - The options used to configure any requests this
+     *                TokenCredential implementation might make.
+     */
+     getToken(scopes?: string | string[], options?: any): Promise<any | null>;
```

Comment:
It's to keep it more loose so that other versions of credentials could be accepted.  For example, `BasicAuthenticationCredentials` returns  `WebResourceLike` whereas the `ServiceClientCredentials` returns `WebResource`.   We originally did that for `signRequest` because different versions of "@azure/ms-rest-js" would change the credential interface.  I figured the same could happen with `getToken`.

Here's an example with the typing screwing up here: https://github.com/Azure/azure-sdk-for-js/issues/10045

Reply:
Is it possible that version of `signRequest` that returns `WebResource` is older? It looks like the current version returns `WebResourceLike`: https://github.com/Azure/ms-rest-js/blob/f6f0d92d79a1dfa8d92ff0891b88bc6b7a349e69/lib/credentials/serviceClientCredentials.ts